src/topology/continuous_function/algebra.lean

```diff
@@ -319,22 +323,65 @@ end subtype
 namespace continuous_map
 
 instance {α : Type*} {β : Type*} [topological_space α] [topological_space β]
-  [semiring β] [topological_semiring β] : semiring C(α, β) :=
+  [non_unital_non_assoc_semiring β] [topological_semiring β] :
+  non_unital_non_assoc_semiring C(α, β) :=
 { left_distrib := λ a b c, by ext; exact left_distrib _ _ _,
```

Comment:
I'm a bit surprised these don't use the injective helpers; maybe I tried that and hit timeouts...